nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/log/kafka/KafkaLogTailer.java

```diff
@@ -0,0 +1,408 @@
+                close();
+            } catch (Exception e1) {
+                log.warn("Error while closing the tailer " + this);
+            }
+            throw new IllegalStateException("poll interrupted because tailer has been closed");
+        }
+        if (log.isDebugEnabled()) {
+            String msg = "Polling " + id + " returns " + records.size() + " records";
+            if (records.size() > 0) {
+                log.debug(msg);
+            } else {
+                log.trace(msg);
+            }
+        }
+        return records.size();
+    }
+
+    @Override
+    public void toEnd() {
+        log.debug("toEnd: " + id);
+        lastOffsets.clear();
+        records.clear();
+        consumer.seekToEnd(Collections.emptyList());
+    }
+
+    @Override
+    public void toStart() {
+        log.debug("toStart: " + id);
+        lastOffsets.clear();
+        records.clear();
+        consumer.seekToBeginning(Collections.emptyList());
+    }
+
+    @Override
+    public void toLastCommitted() {
+        log.debug("toLastCommitted tailer: " + id);
+        String msg = consumer.assignment().stream().map(tp -> String.format("%s-%02d:+%d",
+                getNameForTopic(tp.topic()), tp.partition(), toLastCommitted(tp)))
+                .collect(Collectors.joining("|"));
+        if (msg.length() > 0) {
+            log.info("toLastCommitted offsets: " + group + ":" + msg);
+        }
+        lastCommittedOffsets.clear();
+        lastOffsets.clear();
+        records.clear();
+    }
+
+    protected long toLastCommitted(TopicPartition topicPartition) {
+        Long offset = lastCommittedOffsets.get(topicPartition);
+        if (offset == null) {
+            OffsetAndMetadata offsetMeta = consumer.committed(topicPartition);
+            if (offsetMeta != null) {
+                offset = offsetMeta.offset();
+            }
+        }
+        if (offset != null) {
+            consumer.seek(topicPartition, offset);
+        } else {
+            consumer.seekToBeginning(Collections.singletonList(topicPartition));
+            offset = consumer.position(topicPartition);
+        }
+        if (log.isDebugEnabled()) {
+            log.debug(String.format(" toLastCommitted: %s-%02d:+%d", getNameForTopic(topicPartition.topic()),
+                    topicPartition.partition(),
+                    offset));
+        }
+        return offset;
+    }
+
+    @Override
+    public void seek(LogOffset offset) {
+        log.debug("Seek to: " + offset + " from tailer: " + id);
+        TopicPartition topicPartition = new TopicPartition(prefix + offset.partition().name(),
+                offset.partition().partition());
+        consumer.seek(topicPartition, offset.offset());
+        lastOffsets.remove(topicPartition);
+        List<ConsumerRecord<String, Bytes>> toRemove = records.stream().filter(rec -> rec.partition() == topicPartition.partition()).collect(Collectors.toList());
+        toRemove.forEach(records::remove);
```

Comment:
Use instead `records.removeIf(rec -> rec.partition() == topicPartition.partition())` and probably put `topicPartition.partition()` in a local variable first.